### PR TITLE
Better cloudcare loading

### PR DESCRIPTION
**Pull request by benrudolph**
@czue this was a bit of a pet peeve of mine. this removes some of the loading indicators on CC and replaces it with [nprogress](http://ricostacruz.com/nprogress/). i removed the "All changes saved" since i think the checkmark by the question does that job now. @biyeun this introduces a new loading indicator -- could be useful for other ajax pages where the spinner doesn't work well

before (note the left bar green box and right box):
screen shot 2015-11-21 at 6 21 03 pm
after (note bar on the top):
screen shot 2015-11-21 at 6 16 00 pm

cc: @emord

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/util.js [hunk 1]
```diff
@@ -146,16 +148,12 @@ var hideLoading = function (selector) {
 var tfLoadingComplete = function (isError) {
     hideLoading();
     if (isError) {
-        $('#save-indicator').text(translatedStrings.errSaving).removeClass('alert-warning alert-success').addClass('alert-danger').show();
-    } else {
-        $('#save-indicator').text(translatedStrings.saveAll).removeClass('alert-warning alert-danger').addClass('alert-success').show();
+        showError(translatedStrings.errSaving, $('#cloudcare-notifications'))
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/util.js]: Missing semicolon.

File: corehq/apps/cloudcare/static/cloudcare/js/util.js [hunk 2]
```diff
@@ -146,16 +148,12 @@ var hideLoading = function (selector) {
...
     }
-
 }
 
 var hideLoading = function (selector) {
-    selector = selector || "#loading";
-    $(selector).hide();
+    NProgress.done()
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/util.js]: Missing semicolon.

**Comment by czue** [comment]
whoa that took me a while to spot. super cool. i wonder if @dimagi/product wants to take a peek at this before merging?

**Comment by benrudolph** [comment]
Yeah the screenshot definitely makes it hard to see. I think actually visiting the nprogress site gives you a feel for what it's like. Agree Probably good for product to weigh in

**Comment by amsagoff** [comment]
Will defer to @biyeun  on the progress bar (though agree the current one could definitely use love).

Happy to do some hallway testing too if we want to see whether this is too subtle

**Comment by biyeun** [comment]
@benrudolph my opinion is that this is too subtle. This indicator is not used anywhere else on hq and is dramatically different from what the user would expect to have happen. 

I second @amsagoff for hallway testing to either confirm or disprove my assumptions.

**Comment by benrudolph** [comment]
sure, do you have another suggestion? i don't want to leave it as is for two reasons:
1. the sync now button is going to go where the green box is currently located
2. i have a grand plan of making the cloudcare debugger a retractable box on the right hand side instead of having to scroll up and down every time you want to use it. thus i want to get rid of the spinner thing on the right

**Comment by benrudolph** [comment]
i also think doing an actual hallway test is good. when actually used it seems pretty noticeable, but do agree this is not a pattern we've used on hq

**Comment by czue** [comment]
I'd recommend actually checking this out. I just ran it locally and I think it's a pretty big improvement. you have the little inline spinners and then the thing just kinda of zips across the screen at the top (or crawls across when it's loading). I don't actually think the subtlety thing is a big deal since the whole indicator is sort of a nice to have.

If the reason not to include it is that it's not inline with other HQ stuff, I'd almost advocate for not doing it, but still killing the loading indicator since I don't see what that's adding anymore.

**Comment by amsagoff** [comment]
Oh my god yes to #2!

**Comment by biyeun** [comment]
@benrudolph ok, lets get this in then! can you resolve conflicts?

**Comment by benrudolph** [comment]
updated. also if you feel this isn't a good change, just ping me and i can remove. i've been working on other things so kind of neglected this thread

**Comment by amsagoff** [comment]
B just gave me a demo locally, looks like a big improvement!

**Comment by biyeun** [comment]
@benrudolph just one tiny change. the blue is not the right blue, doesn't block the merge but would be good to change it to #004ebc

**Comment by benrudolph** [comment]
legit, done here: https://github.com/dimagi/touchforms/pull/184/files#diff-3ac2d6c0d21f3005dc08595da3ef93c3 i totally forgot to submit that PR

**Comment by biyeun** [comment]
:+1: waiting on build

**Pull request merged**